src/include/ipc4/base-config.h

```diff
@@ -144,4 +144,19 @@ struct ipc4_output_pin_format {
 	struct ipc4_audio_format audio_fmt; /**< format of the output data */
 } __attribute__((packed, aligned(4)));
 
+struct ipc4_base_module_cfg_ext {
+	/* specifies number of items in input_pins array. Maximum size is 8 */
+	uint16_t nb_input_pins;
+	/* specifies number of items in output_pins array. Maximum size is 8 */
+	uint16_t nb_output_pins;
+	uint8_t reserved[12];
+	/* Specifies format of input pins. Pin format arrays may be non-continuous i.e.
+	 * may contain pin #0 format followed by pin #2 format in case pin #1 will not be
+	 * in use. FW assigned format of the pin based on pin_index, not on a position of
+	 * the item in the array. Applies to both input and output pins.
+	 */
+	struct ipc4_input_pin_format input_pins[1];
+	struct ipc4_output_pin_format output_pins[1];
```

Comment:
This part looks just awful in my view, would it make sense to replace it with
```c
uint8_t pin_formats[];
```
The `output_pins` in itself can not be referenced at all, if nb_input_pins != 1, it is not pointing where it is expected it to be pointing.

Reply:
makes sense. Can you please fix this in this PR too @ujfalusi